12/umbraco-cms/tutorials/getting-started-with-entity-framework-core.md

```diff
@@ -0,0 +1,226 @@
+﻿---
+versionFrom: 12.0.0
+meta.Title: Umbraco Database
+description: A guide to getting started with Entity Framework Core in Umbraco
+---
+
+# Getting started with Entity Framework Core in Umbraco
+
+Using EFCore, it is possible to add custom database tables to your site to store additional data that should not be stored as normal content nodes.
+
+The end result looks like this:
+
+![Database result of a migration](images/db-table.png)
+
+We can then work with this data through Entity framework.
+
+## Using a Composer and Notification Handler
+
+The following code sample shows how this is done using a composer and a notification handler.
+
+With this pattern you create and run a similar migration but trigger it in response to a [notification handler](../fundamentals/code/subscribing-to-notifications.md).
+
+The code for this approach is as follows.
+
+First lets create a BlogComment class.
+
+### BlogComment class
+```
+namespace Umbraco.Demo;
+
+public class BlogComment
+{
+    public int Id { get; set; }
+
+    public Guid BlogPostUmbracoKey { get; set; }
+
+    public required string Name { get; set; }
+
+    public required string Email { get; set; }
+
+    public required string Website { get; set; }
+
+    public string Message { get; set; } = string.Empty;
+}
+```
+
+Now that we have the model, lets create a `DbContext` class so we can interact with the database, we'll call this `BlogContext`.
+
+### BlogContext class
+```
+using Microsoft.EntityFrameworkCore;
+
+namespace Umbraco.Demo;
+
+public class BlogContext : DbContext
+{
+    public BlogContext(DbContextOptions<BlogContext> options)
+        : base(options)
+    {
+    }
+
+    public required DbSet<BlogComment> BlogComments { get; set; }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder) =>
+        modelBuilder.Entity<BlogComment>(entity =>
+        {
+            entity.ToTable("blogComment");
+            entity.HasKey(e => e.Id);
+            entity.Property(e => e.Id).HasColumnName("id");
+            entity.Property(e => e.BlogPostUmbracoId).HasColumnName("blogPostUmbracoKey");
+            entity.Property(e => e.Message).HasColumnName("message");
+            entity.Property(e => e.Website).HasColumnName("website");
+            entity.Property(e => e.Email).HasColumnName("email");
+        });
+}
+```
+
+We need to register this DbContext to use it, in Umbraco because we need to register a few other things to have Scopes working, we have a helpful extension method you use:
```

Comment:
```suggestion
### Step 3: Register the DbContext

We need to register this DbContext to use it, in Umbraco because we need to register a few other things to have Scopes working, we have a helpful extension method you use:
```